x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -243,13 +243,12 @@ public boolean isAvailableWithLicense(XPackLicenseState licenseState) {
         }
 
         // The model license does not matter, this is the highest licensed level
-        if (licenseState.isActive() && XPackLicenseState.isAllowedByOperationMode(
-            licenseState.getOperationMode(), License.OperationMode.PLATINUM, true)) {
+        if (licenseState.isAllowedByLicense(License.OperationMode.PLATINUM, true, true)) {
```

Comment:
Address feedback https://github.com/elastic/elasticsearch/pull/51864#discussion_r376830829
Single call to remove potential race condition.